pkg/database/authorized_app_stats.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// AuthorizedAppStats represents statistics related to a user in the database.
+type AuthorizedAppStats struct {
+	gorm.Model
+	Date            time.Time `gorm:"unique_index:idx_date_app_realm"`
```

Comment:
I can't find any docs on what this actually does? Is it valid to have multiple fields against the same index?

Reply:
It's a composite unique index to ensure each entry per user/realm/date is unique, since this entry is really a wrapup for the day. I am trying to find docs now, but it is that with a specified name for the index.